### ISSUE TITLE
[Java] Type Format Ignored on Simple Type Definitions

### ISSUE THREAD

**Issue by dtgkeithmarsh** [swagger-api/swagger-codegen#7912]
##### Description

Type format appears to be ignored on primitive typedefs.  I can't see anything in the documentation that prevents a type being declared that maps onto a simple primitive type.  If you use this to strongly type your interface, it appears the format is ignored at code generation time.

##### Swagger-codegen version

2.3.1 and 2.4.0

##### Swagger declaration file content or url

```yaml
swagger: "2.0"
info:
  title: "Format Ignored?"
  version: "1.0"
paths:
  /demo:
    post:
      consumes: []
      parameters:
      - name: "body"
        in: "body"
        required: true
        schema:
          $ref: "#/definitions/Demo"
      responses:
        200:
          description: "Status 200"
definitions:
  Demo:
    type: "object"
    properties:
      bad:
        $ref: "#/definitions/UnixTime"
      good:
        type: integer
        format: "int64"
  UnixTime:
    type: "integer"
    format: "int64"
```

##### Command line used for generation

    java -jar swagger-codegen-cli-2.4.0-20180325.152919-192.jar generate -l jaxrs -i demo.yaml -o demo

##### Steps to reproduce

Run the command.  It reproduces on 2.4.0 and 2.3.1 jaxrs and java.  3.0.0-rc0 fails to generate because model.mustache not found.   The problem is for any format.  A number with a format float behaves similarly, generating Float for in place format and BigDecimal for the child type def.

The generated code for the supplied yaml is

```java
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaJerseyServerCodegen", date = "2018-03-26T17:10:11.467+01:00")
public class Demo   {
  @JsonProperty("bad")
  private Integer bad = null;

  @JsonProperty("good")
  private Long good = null;
...
```

##### Related issues/PRs

I couldn't find anything similar

**Comment by jmini** [comment]
I think this is a duplicate of #7754, it should be fixed with `3.0.0-SNAPSHOT`

**Comment by dtgkeithmarsh** [comment]
Thanks.  I missed that.
